V2 loop parallelism dependencies

gmfrasca opened the pull request:
Simulation of a PR to upstream

amadhusu reviewed (approved):
Looks good comparing the PR upstream and the requested changes made here.

gmfrasca committed:
feat(backend): Add Parallelism Limit to ParallelFor tasks

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

gmfrasca committed:
feat(backend): Add intermediate Template for iterator Tasks in ArgoCompiler

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

gmfrasca committed:
test: Add argoCompiler test case to validate individual parallel-limited tasks

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

gmfrasca committed:
test: Update tests for ParallelFor loop update

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

gmfrasca committed:
WIP: Fix broken dependtTasks in ParallelFor

gmfrasca committed:
fix(backend): pass correct ParentDagID to iterator DAG

- Passthrough ParentDagID rather than DriverExecutionID to iterator such
  that iteration item correctly detects dependentTasks.
- Remove depends from iterator DAG as it is already handled by
  root-level task
- Update Iterator template names/nomenclature for clarity
- Update tests accordingly

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>
